src/app/App.tsx

```diff
@@ -1,50 +1,61 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { BrowserRouter as Router } from 'react-router-dom';
 import { I18nextProvider } from 'react-i18next';
-import { ApolloClient, InMemoryCache, ApolloProvider } from '@apollo/client';
 import '@patternfly/react-core/dist/styles/base.css';
 import { AppLayout, Loading } from '@app/components';
 import { AppRoutes } from '@app/routes';
 import { initI18N } from '@i18n/i18n';
 import { KeycloakAuthProvider, KeycloakContext, getKeycloakInstance } from '@app/auth/keycloak';
 import '@app/app.css';
+import { useVerifyTokenQuery } from '@app/models';
 
 let keycloak: Keycloak.KeycloakInstance | undefined;
+let tokenresponse: string | boolean | unknown;
 
 const App: React.FunctionComponent = () => {
   const [initialized, setInitialized] = useState(false);
-  const client = new ApolloClient({
-    uri: process.env.GRAPHQL_ENDPOINT || 'http://localhost:8080/graphql',
-    cache: new InMemoryCache(),
+  const location = window.location;
+  const params = new URLSearchParams(location.search);
+  const token = params.get('token') as string;
+  const email = params.get('email') as string;
+  const { loading, error, data } = useVerifyTokenQuery({
+    variables: {
+      emailId: email,
+      token: token,
+    },
+    fetchPolicy: 'cache-and-network',
+    notifyOnNetworkStatusChange: true,
   });
-
-  //Initialize the client
-  React.useEffect(() => {
+  useEffect(() => {
     const init = async () => {
       keycloak = await getKeycloakInstance();
+      if (!loading && token && email) {
```

Comment:
Can we improve checks for `setInitialized` state. 
Below we are again updating state `setInitialized` as true. In this case above check will skip and this check doesn't make sense.